### ISSUE TITLE
"Too much recursion" error FF4

### ISSUE THREAD

**Issue by WraithKenny** [peol/jquery.imgloaded#2]
When the plugin is run in FF4 this error is reported and the rest of the scripts on the page don't run.

**Comment by WraithKenny** [comment]
See the comments on api.jquery.com/load-event/ for solution/alternative.

**Comment by peol** [comment]
I think I got that fixed locally, but I got issues replicating the image caching bugs in IE at the moment, so I'll wait to push it to github before actually verifying everything's working.

**Comment by peol** [comment]
Yes, as of now, you should be using either bind-event-before-setting-source approach, or the workaround on the API site. This plugin is right now busted without a clear solution, I'm afraid.

**Comment by jcljules** [comment]
The problem with this function is that it doesn't just bind to images - it adds this loader to every single element on the page. Change event.special.load to event.special.imgpreload, then bind an imgpreload event to the specific images you'd like to preload. Worked like a charm for me, in all browsers.

**Comment by chrihala** [comment]
jcljules's advice seems to work in Firefox 3.6 and Chrome 7. Thanks mate!

**Comment by jefferyto** [comment]
I have an experimental fork of this plugin at https://github.com/jefferyto/jquery.imgloaded/tree/pi . Can you give it a try? I'd like to get some feedback before submitting a pull request. Thanks!
